### DISCUSSION TITLE
What is the proper way to deal with merging branches when each branch has separate releases?

### DISCUSSION THREAD

**Discussion by suspiration** [semantic-release/semantic-release#1854] [Q&A]
My branches config is as follow
`"branches": [
        "master",
        {"name": "development", "channel": "beta", "prerelease": "beta"}
    ]`

development branch is setup to automatically generate a prerelease when new changes are pushed.  When I try to merge development into master, I'm getting conflict on CHANGELOG.MD, package.json and package-lock.json since both master and development releases have updated those files.  

What is the proper way to handle this?  Any help is greatly appreciated!

**Comment by bigbrozer** [comment]
At least for `CHANGELOG.md`, there is [an interesting approach](https://github.com/semantic-release/changelog/issues/51#issuecomment-682609394) by using a "dynamic" config file for semantic-release. It permits to generate, for example, a CHANGELOG for specific branches or disable it for pre-releases, and so on... this simplify a bit the git merging process.
